src/Temporalio/Bridge/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+#![allow(
+    // We choose to have narrow "unsafe" blocks instead of marking entire
+    // functions as unsafe. Even the example in clippy's docs at
+    // https://rust-lang.github.io/rust-clippy/master/index.html#not_unsafe_ptr_arg_deref
+    // cause a rustc warning for unnecessary inner-unsafe when marked on fn.
+    // This check only applies to "pub" functions which are all exposed via C
+    // API.
+    clippy::not_unsafe_ptr_arg_deref,
+)]
+
+pub mod client;
+pub mod runtime;
+pub mod testing;
+
+use std::collections::HashMap;
+
+#[repr(C)]
+pub struct ByteArrayRef {
+    data: *const u8,
+    size: libc::size_t,
+}
+
+impl ByteArrayRef {
```

Comment:
Earlier versions of these impls were all sorts of `AsRef`, `From`, `TryFrom`, etc impls. But it became a bit confusing what was and wasn't and what exact conversions were happening, so I moved it back to explicit conversions which clarified everything at a minor loss of ergonomics.